lib/class/vainfo.class.php

```diff
@@ -804,7 +804,20 @@ private function _cleanup_id3v2($tags)
 
             switch ($tag) {
                 case 'genre':
-                    // Pass the array through
+                    // read additional id3v2 delimiters from config
+                    $delimiters = (array) AmpConfig::get('additional_id3v2_genre_delimiters');
+                    if (isset($data) && is_array($data) &&
+                        isset($delimiters) && is_array($delimiters)) {
+                        // if there is only one element in the array,
+                        // iterate through the delimiters
+                        // until the genre string splits into several elements
+                        $i = 0;
+                        while (count($data) === 1 && $i < count($delimiters)) {
```

Comment:
Maybe we could replace this whole while thing with a simple preg_split() after checking $data?
There we could also filter the whitespace from the genre elements (multiple genre in beets are like this: "genre1, genre2, ...")
This would also fix the problem that it cannot handle comma. Of course then the delimiters should not be parsed to an array.